### PR TITLE
Download and install kubectl/aws-iam-auth

### PR DESCRIPTION
**Pull request by brunomanzo**
Download kubectl/aws-iam-auth and put on path

### CODE REVIEW

File: tasks/client.yml [hunk 1]
```diff
@@ -2,3 +2,44 @@
...
+  get_url:
+    url: https://amazon-eks.s3-us-west-2.amazonaws.com/1.11.5/2018-12-06/bin/linux/amd64/kubectl
+    dest: /opt/k8s/kubectl
+    mode: 0775
+  when: '"not found" in kubectl_path.stdout'
+
+- name: Create link to kubectl
+  file:
+    src: /opt/k8s/kubectl
+    dest: /usr/local/bin/kubectl
```
Comment (weslleycamilo) [inline tasks/client.yml]: @brunomanzo  why not download it directly to its path? instead of doing a link ?

**Review by weslleycamilo — CHANGES_REQUESTED** [review]

File: tasks/client.yml [hunk 2]
```diff
@@ -2,3 +2,44 @@
...
+
+- name: Download aws iam auth
+  get_url:
+    url: https://amazon-eks.s3-us-west-2.amazonaws.com/1.11.5/2018-12-06/bin/linux/amd64/aws-iam-authenticator
+    dest: /opt/k8s/aws-iam-authenticator
+    mode: 0775
+  when: '"not found" in aws_iam_path.stdout'
+
+- name: Create link to aws iam auth
+  file:
```
Comment (weslleycamilo) [inline tasks/client.yml]: The same here.. why not download directly the aws-iam-autheticator?

File: tasks/client.yml [hunk 3]
```diff
@@ -2,3 +2,44 @@
...
 ## aws-iam-authenticator
+
+- name: "Get kubectl path"
+  shell: "which kubectl || echo 'not found'"
+  register: kubectl_path
+
+- name: Create bin directory
+  file:
+    path: /opt/k8s
+    state: directory
```
Comment (mtulio) [inline tasks/client.yml]: @brunomanzo the directory `HOME/bin` is recommended. `/opt` should have extra permissions that is not the scope of this role.

File: tasks/client.yml [hunk 4]
```diff
@@ -2,3 +2,44 @@
...
+  when: '"not found" in kubectl_path.stdout'
+
+- name: "Get aws iam auth path"
+  shell: "which aws-iam-authenticator || echo 'not found'"
+  register: aws_iam_path
+
+- name: Download aws iam auth
+  get_url:
+    url: https://amazon-eks.s3-us-west-2.amazonaws.com/1.11.5/2018-12-06/bin/linux/amd64/aws-iam-authenticator
+    dest: /opt/k8s/aws-iam-authenticator
```
Comment (mtulio) [inline tasks/client.yml]: Same commented above about the BIN root path.

File: tasks/main.yml [hunk 5]
```diff
@@ -1,7 +1,8 @@
 ---
 - include: client.yml
   when: eks_clients_install
-
+  become: yes
```
Comment (mtulio) [inline tasks/main.yml]: Please, leave default as `no`, in general the playbook should not have privilated access to install kube clients or its bin dependencies in your local env (laptop)

**Review by mtulio — CHANGES_REQUESTED** [review]

File: tasks/client.yml [hunk 6]
```diff
@@ -2,3 +2,44 @@
...
+  register: kubectl_path
+
+- name: Create bin directory
+  file:
+    path: /opt/k8s
+    state: directory
+
+- name: Download kubectl
+  get_url:
+    url: https://amazon-eks.s3-us-west-2.amazonaws.com/1.11.5/2018-12-06/bin/linux/amd64/kubectl
```
Comment (weslleycamilo) [inline tasks/client.yml:12]: It would be a var so we can update it time to time .. what do you think ?

File: tasks/client.yml [hunk 7]
```diff
@@ -2,3 +2,25 @@
...
+    mode: 0775
+  when: '"not found" in kubectl_path.stdout'
+
+- name: "Get aws iam auth path"
+  shell: "which aws-iam-authenticator || echo 'not found'"
+  register: aws_iam_path
+
+- name: Download aws iam auth
+  get_url:
+    url: https://amazon-eks.s3-us-west-2.amazonaws.com/1.11.5/2018-12-06/bin/linux/amd64/aws-iam-authenticator
```
Comment (mtulio) [inline tasks/client.yml:23]: If you accept the Wessly suggestion above, you also could add this URL as a var.

**Pull request merged**